### PR TITLE
IVY-3602-create-headless-sample-shop

### PR DESCRIPTION
**Pull request by cdans**

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -410,12 +421,97 @@ router.post('/checkout', async (req, res) => {
...
+      },
+    }
+
+    const { data: session } = await axios.post(
+      `${config.IVY_API_URL}/service/checkout/session/create`,
+      data,
+      {
+        headers: {
+          'X-Ivy-Api-Key':
+            config.IVY_API_KEY ?? '6540c302881f7116cb64c284.246897d2-7953-4a10-afdb-0707391acb5f',
```
Comment (kostoman) [inline src/routes/index.js:491]: no hardcoded apikeys

File: src/routes/index.js [hunk 2]
```diff
@@ -410,12 +421,97 @@ router.post('/checkout', async (req, res) => {
...
+        headers: {
+          'X-Ivy-Api-Key': config.IVY_API_KEY,
+        },
+      },
+    )
+
+    const checkoutInterval = setInterval(
+      () => checkCheckoutStatus(session.id, checkoutInterval),
+      5000,
+    )
```
Comment (kostoman) [inline src/routes/index.js:511]: i would not do this hear, instead call this endpoint from the frontend with a interval

File: package.json [hunk 3]
```diff
@@ -26,7 +26,8 @@
     "express": "4.18.2",
     "express-async-errors": "3.1.1",
     "morgan": "1.10.0",
-    "pug": "3.0.2"
+    "pug": "3.0.2",
+    "uuid": "^9.0.1"
```
Comment (kostoman) [inline package.json:30]: isn't there a uuid from the crypto node package already existing? like: `import { randomUUID } from 'node:crypto'`

File: src/routes/index.js [hunk 4]
```diff
@@ -410,12 +421,97 @@ router.post('/checkout', async (req, res) => {
...
             reqData.custom === 'true' ? req.app.get('customApiKey') : config.IVY_API_KEY,
         },
-      }
+      },
     )
 
     console.log('session:', session)
     res.json({
       url: session.redirectUrl,
+      test: true,
```
Comment (kostoman) [inline src/routes/index.js:430]: whats `test: true`?

File: src/routes/index.js (hunk 1 above)
Comment (cdans) [reply src/routes/index.js:491]: this was by accident, was a bit late yesterday evening ...

removed it

File: src/routes/index.js (hunk 2 above)
Comment (cdans) [reply src/routes/index.js:510]: isn't this more a backend topic?
if the shop reloads the interval would be canceled

Comment (kostoman) [reply src/routes/index.js:510]: but i dont even get what this interval should be? maybe you need to explain it to me

File: package.json (hunk 3 above)
Comment (cdans) [reply package.json]: yep, true

changed it

File: src/routes/index.js (hunk 4 above)
Comment (cdans) [reply src/routes/index.js]: removed it, was some testing stuff

File: src/routes/index.js (hunk 2 above)
Comment (cdans) [reply src/routes/index.js]: So the app knows it the payment was successful and can redirect to the success-page

**Review by FilipKajetaniak — APPROVED** [review]

**Pull request merged**